src/index.js

```diff
@@ -76,17 +75,37 @@ async function updatePackageJSON(pkg, options) {
   return pkg;
 }
 
+function prettyPrint(json) {
+  return JSON.stringify(json, null, 2);
+}
+
+function showPatch(filename, before, after) {
+  console.log(
+    diff.createPatch(
+      filename,
+      before,
+      after,
+      "Before Upgrade",
+      "After Upgrade"
+    )
+  );
+  console.log("");
+}
+
 async function writePackageJSON(options) {
   let { pkg, path } = await readPkgUp({ normalize: false });
-
+  let oldPkg = prettyPrint(pkg);
   pkg = await updatePackageJSON(pkg, options);
 
   if (pkg.babel) {
     console.log("Updating package.json 'babel' config");
     pkg.babel = upgradeConfig(pkg.babel, options);
   }
+  showPatch("package.json", oldPkg, prettyPrint(pkg));
 
-  await writeJsonFile(path, pkg, { detectIndent: true });
+  if (!options.dryRun) {
+    await writeJsonFile(path, pkg, { detectIndent: true });
```

Comment:
for testing this, perhaps we could stub the default export of `write-json-file` with jest and then assert that it is called (with the correct arguments) when `options.dryRun` is true and assert that it is not called when `options.dryRun` isn't?

Reply:
I was thinking along the same lines. I can get started on that. What about the diffs? Is there any value to testing the diff output itself?